src/main/java/io/redlink/more/data/component/SchedulingComponent.java

```diff
@@ -0,0 +1,37 @@
+package io.redlink.more.data.component;
+
+import io.redlink.more.data.properties.GarminProperties;
+import io.redlink.more.data.service.GarminService;
+import jakarta.annotation.PostConstruct;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.scheduling.TaskScheduler;
+import org.springframework.stereotype.Component;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+
+@Component
+public class SchedulingComponent {
+    private static final Logger LOG = LoggerFactory.getLogger(SchedulingComponent.class);
+    private final TaskScheduler scheduler;
+    private final GarminService garminService;
+    private final GarminProperties garminProperties;
+
+    public SchedulingComponent(TaskScheduler scheduler, GarminService garminService, GarminProperties garminProperties) {
+        this.scheduler = scheduler;
+        this.garminService = garminService;
+        this.garminProperties = garminProperties;
+    }
+
+    @PostConstruct
```

Comment:
I would use

```suggestion
    @EventListener(ApplicationReadyEvent.class)
```
instead as this is after the application is started